### PR TITLE
Refactoring/test info

### PR DESCRIPTION
**Pull request by danymill**

### CODE REVIEW

**Review by dzmitrymishyn — APPROVED** [review]

**Review by MitskevichVS — APPROVED** [review]

File: client/app/containers/test-run-info/elastic-search/elasticsearch.service.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    let esSettings;
+
+    return {
+        initService,
+        fetchCount,
+        fetchSearch,
+    };
+
+    function buildMustClause(termsArray) {
+        return termsArray.map((searchField) => ({ 'term': searchField }));
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/elastic-search/elasticsearch.service.js:20]: here

File: client/app/containers/test-run-info/elastic-search/elasticsearch.service.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        }
+    }
+
+    function fetchCount(esIndex, searchFields, canceler) {
+        const params = {
+            query: {
+                bool: {
+                    must: buildMustClause(searchFields),
+                }
+            }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/elastic-search/elasticsearch.service.js:76]: here

File: client/app/containers/test-run-info/elastic-search/elasticsearch.service.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        };
+        const settings = {
+            headers: { Authorization:  esSettings.authBasic},
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/elastic-search/elasticsearch.service.js:79]: here

File: client/app/containers/test-run-info/elastic-search/elasticsearch.service.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        return $http.post(`${esSettings.url}/${esIndex}/_count`, params, settings)
+            .then((response) => response.data);
+    }
+
+    function fetchSearch(esIndex, searchFields, from, size, canceler) {
+        const params = {
+            query: {
+                bool: {
+                    must: buildMustClause(searchFields),
+                }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/elastic-search/elasticsearch.service.js:92]: here

File: client/app/containers/test-run-info/elastic-search/elasticsearch.service.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            },
+            sort: [{
+                'timestamp': {
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/elastic-search/elasticsearch.service.js:95]: here

File: client/app/containers/test-run-info/elastic-search/elasticsearch.service.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                    order: 'asc',
+                },
+            }],
+            size,
+            from,
+        };
+        const settings = {
+            headers: { Authorization:  esSettings.authBasic},
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/elastic-search/elasticsearch.service.js:103]: here

File: client/app/containers/test-run-info/logs-agents/default.logs-agent.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            log.message = 'Screenshot is captured';
+
+            const relativeImageUrl = log.originalMessage
+                .replace(new RegExp(`^(${log.tenant || authService.tenant})`, 'i'), '');
+
+            if (relativeImageUrl) {
+                const imageUrl = `${authService.serviceUrl}${relativeImageUrl}`;
+                const thumbnailUrl = imageUrl.replace(fileExtensionPattern, '_thumbnail$1');
+
+                log.urls.image = { path: imageUrl, name: 'screenshot', }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/logs-agents/default.logs-agent.js:37]: it's bad to add comma after the name property. Only if you use new line for each property

File: client/app/containers/test-run-info/test-run-info.controller.js [hunk 8]
```diff
@@ -1,206 +1,128 @@
...
-            logGetter: {
-                from: null,
-                pageCount: page,
-                getSizeFunc: function (count) {
-                    return count;
-                },
-                accessFunc: null
-            }
+            from: 0,
         }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/test-run-info.controller.js:104]: here

File: client/app/containers/test-run-info/test-run-info.controller.js [hunk 9]
```diff
@@ -210,436 +132,127 @@ const testRunInfoController = function testRunInfoController(
...
-        log.showMore = !log.showMore;
-        if (!log.showMore) {
-            $timeout(function () {
-                if (scrollableElem.scrollTop > rowElem.offsetTop) {
-                    scrollableElem.scrollTop = rowElem.offsetTop;
-                }
-            }, 0);
-        }
-    }
+        const testCopy = {...test};
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/test-run-info.controller.js:139]: here

File: client/app/containers/test-run-info/test-run-info.controller.js [hunk 10]
```diff
@@ -210,436 +132,127 @@ const testRunInfoController = function testRunInfoController(
...
+                bindToController: true,
+                parent: angular.element(document.body),
+                targetEvent: event,
+                clickOutsideToClose: true,
+                fullscreen: false,
+                escapeToClose: false,
+                locals: {
+                    test: vm.test,
+                    activeArtifactId: activeArtifact.id,
                 }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/test-run-info.controller.js:204]: here

File: client/app/containers/test-run-info/test-run-info.controller.js [hunk 11]
```diff
@@ -1058,50 +432,192 @@ const testRunInfoController = function testRunInfoController(
...
-        driversCount = $scope.drivers.length;
-        postModeConstruct(test, vm.pageSessionId);
+    function initLiveMode() {
+        addDrivers(getLiveVideoArtifacts(vm.test.artifacts, true));
+        getLiveESLogs$()
+            .pipe(
+                takeUntil(logsGettingDestroy$),
+            )
+            .subscribe({
+                complete() { vm.isLogsLoading = false; }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/test-run-info.controller.js:442]: here

File: client/app/containers/test-run-info/test-run-info.controller.js [hunk 12]
```diff
@@ -1058,50 +432,192 @@ const testRunInfoController = function testRunInfoController(
...
-        addDrivers(videoArtifacts);
-        postModeConstruct(test, vm.pageSessionId);
+    function initRecordMode() {
+        addDrivers(getVideoArtifacts(vm.test.artifacts, true));
+        fiveMinsLogsFetcher$()
+            .pipe(
+                takeUntil(logsGettingDestroy$),
+            )
+            .subscribe({
+                complete() { vm.isLogsLoading = false; }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/test-run-info.controller.js:453]: here

File: client/app/containers/test-run-info/test-video-slider/test-video-slider.controller.js [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+        const prevDriver = this.drivers[e.previousIndex];
+
+        $timeout(() => {
+            handlePrevDriver(prevDriver);
+
+            this.onDriverChange({
+                $drivers: {
+                    prevDriverIndex: e.previousIndex,
+                    activeDriverIndex: e.activeIndex,
+                }
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/test-video-slider/test-video-slider.controller.js:44]: here

File: client/app/containers/test-run-info/test-video-slider/test-video-slider.controller.js [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+            });
+        }, 300); // 300 - is slider's default transition duration
+
+
```
Comment (DmitryMishin) [inline client/app/containers/test-run-info/test-video-slider/test-video-slider.controller.js:48]: here

**Review by dzmitrymishyn — CHANGES_REQUESTED** [review]

**Review by dzmitrymishyn — APPROVED** [review]

**Pull request merged**